types/src/transaction/authenticator.rs

```diff
@@ -90,70 +97,162 @@ impl TransactionAuthenticator {
         }
     }
 
-    /// Return bytes for (self.public_key | self.scheme_id). To create an account authentication
-    /// key, sha3 these bytes
-    // TODO: move AuthenticationKey to this file and make this private
-    pub fn compute_authentication_key_preimage(
-        mut public_key_bytes: Vec<u8>,
-        scheme_id: u8,
-    ) -> AuthenticationKeyPreimage {
-        public_key_bytes.push(scheme_id);
-        AuthenticationKeyPreimage(public_key_bytes)
+    /// Return an authentication key preimage derived from `self`'s public key and scheme id
+    pub fn authentication_key_preimage(&self) -> AuthenticationKeyPreimage {
+        AuthenticationKeyPreimage::new(self.public_key_bytes().to_vec(), self.scheme())
     }
 
-    // TODO: move AuthenticationKey to this file to avoid try_from
-    fn preimage_to_authentication_key(preimage: &AuthenticationKeyPreimage) -> AuthenticationKey {
-        AuthenticationKey::try_from(HashValue::from_sha3_256(&preimage.0).to_vec()).unwrap()
+    /// Return an authentication key derived from `self`'s public key and scheme id
+    pub fn authentication_key(&self) -> AuthenticationKey {
+        AuthenticationKey::from_preimage(&self.authentication_key_preimage())
     }
+}
 
-    pub fn authentication_key_preimage(&self) -> AuthenticationKeyPreimage {
-        Self::compute_authentication_key_preimage(
-            self.public_key_bytes().to_vec(),
-            self.scheme_id(),
-        )
+/// A struct that represents an account authentication key. An account's address is the last 16
+/// bytes of authentication key used to create it
+#[derive(Debug, Ord, PartialOrd, Eq, PartialEq, Hash, Clone, Copy, CryptoHasher)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+pub struct AuthenticationKey([u8; AUTHENTICATION_KEY_LENGTH]);
+
+impl AuthenticationKey {
+    pub const fn new(bytes: [u8; AUTHENTICATION_KEY_LENGTH]) -> Self {
+        Self(bytes)
+    }
+
+    pub fn random() -> Self {
```

Comment:
Add some doc comments on every new public function. Actually, I though CI captures that.